elyra/pipeline/component_parser_kfp.py

```diff
@@ -140,11 +167,67 @@ def _get_adjusted_parameter_fields(self,
                 for command in component_body['implementation']['container']['command']:
                     if isinstance(command, dict) and list(command.values())[0] == io_object_name and \
                             list(command.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
+                        adjusted_type = f"{io_object_type}Path"
             if "args" in component_body['implementation']['container']:
                 for arg in component_body['implementation']['container']['args']:
                     if isinstance(arg, dict) and list(arg.values())[0] == io_object_name and \
                             list(arg.keys())[0] == f"{io_object_type}Path":
-                        adjusted_type = "file"
+                        adjusted_type = f"{io_object_type}Path"
 
         return adjusted_type
+
+    def determine_type_information(self, parsed_type: str) -> Tuple[str, str, str, bool, Any]:
+        """
+        Takes the type information of a component parameter as parsed from the component
+        specification and returns a new type that is one of several standard options.
+
+        """
+        type_lowered = parsed_type.lower()
+        type_options = ['dictionary', 'dict', 'set', 'list', 'array', 'arr', 'inputpath', 'inputvalue', 'outputpath']
+
+        # Prefer types that occur in a clause of the form "[type] of ..."
+        # E.g. "a dictionary of key/value pairs" will produce the type "dictionary"
+        for option in type_options:
+            if any(word + " of " in type_lowered for word in type_options):
+                reg = re.compile(f"({option}) of ")
+                match = reg.search(type_lowered)
+                if match:
+                    type_lowered = option
+                    break
+            elif option in type_lowered:
+                type_lowered = option
+                break
```

Comment:
1. What is the purpose of the nested for loop across `type_options`?  Isn't the outer loop sufficient?  E.g.,
```python
if f"{option} of" in type_lowered:
```

2. Same goes for the regular expression stuff.  If the 'of' clause is in the parsed type comment don't we know `option` by virtue of the for loop?

3. Regarding the if/elif here.  If `{option} of` is in `type_lowered` then `{option}` is in `type_lowered`, so isn't it sufficient to only ask the latter?

Reply:
Just for some context, I think Sourcery refactored this clause from my initial implementation, leading to it needing some refinement like you mentioned. 

The only thing I would point out regarding bullet 3 is that the `elif` catches cases where ` of` doesn't exist in `type_lowered`, so I believe we still need it. E.g., to determine data types from descriptions such as `this parameter is an array`

Reply:
Thanks for the response Kiersten - that's helpful.

Regarding the 3rd item, I guess my point is that ONLY the `elif` clause is necessary since it's actually a subset of the `if` clause that checks for `of`.  That is, if `option of` is in `type_lowered` then `option` is in `type_lowered`, so isn't only the `elif` portion of those checks sufficient?

I feel like I'm missing something but can't see it.

Reply:
Ah yes, I see what you're saying! Yes, you're right. I agree that I feel like I might be missing some obscure case, but I think that's just a side effect of this Sourcery refactor making me second guess myself.